File: dataset/human/java/file_6307.java
import java.io.*;
import java.util.*;

final class Solver {
	static final Solver INSTANCE = new Solver();

	void solve(Iterator<String> sc) {
		int K = Integer.parseInt(sc.next());
		int S = Integer.parseInt(sc.next());
		int sum = 0;
		for (int x = 0; x <= K && x <= S; x++) {
			int remaining = S - x;
			sum += Math.max(Math.min(remaining, K) - Math.max(remaining - K, 0) + 1, 0);
		}
		System.out.println(sum);
	}
}

class Main {
	public static void main(String... args) {
		BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
		Solver.INSTANCE.solve(new Iterator<String>() {
			private StringTokenizer tokenizer;

			public boolean hasNext() {
				return true;
			}

			public String next() {
				if (tokenizer == null || !tokenizer.hasMoreTokens()) {
					try {
						tokenizer = new StringTokenizer(reader.readLine());
					} catch (IOException e) {
						throw new UncheckedIOException(e);
					}
				}
				return tokenizer.nextToken();
			}
		});
	}
}

File: dataset/human/c/file_6307.c
#include<stdio.h>
int main()
{
  int i,n,r[200000],max,min;

  scanf("%d",&n);
  for(i=0; i<n; i++) scanf("%d",&r[i]);
  
  min = r[0];
  max = r[1]-r[0];
  for(i=1; i<n; i++) {
    if(max < r[i]-min) max = r[i] - min;
    if(min > r[i]) min = r[i];
  }

  printf("%d\n",max);
  
  return 0;
}
